internal/pathwatcher/utils.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+// Package pathwatcher provides helper functions for creating file and directory watchers
+package pathwatcher
+
+import (
+	"github.com/fsnotify/fsnotify"
+	"github.com/open-policy-agent/opa/loader"
+)
+
+// CreatePathWatcher creates watchers to monitor for path changes
+func CreatePathWatcher(rootPaths []string) (*fsnotify.Watcher, error) {
```

Comment:
From some manual testing, it doesn't seem like `rootPaths` is fully respected. E.g. if I run `opa test policy.rego -w` and add/modify a file in the same directory as `policy.rego` but _not_ `policy.rego`, I still trigger a new test cycle.

I'm confident this is an existing issue in `opa run`, but I'm unsure we can get away with also having this issue (and other filewatcher-related issues) in watchmode for `test`, which is a new feature ..

Reply:
This particular issue might be in `fsnotify`, though; and then there's probably little we can do.

Reply:
> From some manual testing, it doesn't seem like rootPaths is fully respected. E.g. if I run opa test policy.rego -w and add/modify a file in the same directory as policy.rego but not policy.rego, I still trigger a new test cycle.

Correct, this is an `fsnotify` issue. They don't recommend watching individual files.

Also if you do `opa test policy.rego -w` and modify `policy.rego`, you'll get a loading error like `Load error:  1 error occurred during loading: stat policy.rego: no such file or directory`.  Per the `fsnotify` documentation:

```
// # Watching files
//
// Watching individual files (rather than directories) is generally not
// recommended as many tools update files atomically. Instead of "just" writing
// to the file a temporary file will be written to first, and if successful the
// temporary file is moved to to destination removing the original, or some
// variant thereof. The watcher on the original file is now lost, as it no
// longer exists.
//
// Instead, watch the parent directory and use Event.Name to filter out files
// you're not interested in.
```

Reply:
This makes me think that we should print a warning when running watch mode and the user has specified individual files ..
For both `test` and `run`.

Reply:
Updated the helper messages of both `test` and `run` to include a note about this.